bol/openapi/api.py

```diff
@@ -26,6 +26,14 @@ def __init__(self, api):
     def products(self, product_ids):
         path = 'products/' + ','.join(product_ids)
         return self.request('GET', path)
+        
+    def search(self, query):
+        """
+        query might be 'Harry Potter', 'an_EAN' or 'an_ISBN'. 
+        For exact search, use extra quotation marks, for example: '"Harry Potter"'.
+        """
+        path = 'search/?q=' + query
```

Comment:
Here, the query params are not properly URL encoded. The `self.request` needs to be extended to accept a `params` parameter... then, simply do `self.request('GET', path, params={'q': query})`